backend/src/routes/index.ts

```diff
@@ -121,6 +139,12 @@ const routes: FastifyPluginAsync = async (server) => {
         switch (mode.toLowerCase()) {
           case 'sponsor': {
             const date = new Date();
+            const provider = new providers.JsonRpcProvider(networkConfig.bundler);
+            const signer = new Wallet(privateKey, provider)
+            if (txnMode) {
+              const IndexerData = await getIndexerData(await signer.getAddress(), userOp.sender, date.getMonth(), date.getFullYear(), noOfTxns, indexerEndpoint);
```

Comment:
Nested awaits on the same line, this is complicated and make debugging difficult. Please split this out and remember to catch any errors